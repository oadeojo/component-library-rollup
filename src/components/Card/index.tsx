export { default as Card } from './Card';
export { Props } from './Card';
