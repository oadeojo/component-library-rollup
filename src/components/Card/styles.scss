.card-container {
    * + * {
        margin-top: 20px;
    }

    &.card {
        max-width: 350px;
    }
}

.card {
    background-color: #F7F7F7;
    /* just in case there no content*/
    padding: 20px 20px;
    margin: 0 auto 25px;
    margin-top: 50px;
    /* shadows and rounded borders */
    border-radius: 2px;
    box-shadow: 0px 2px 2px rgba(0, 0, 0, 0.3);

    &__profile-img {
        height: 96px;
        margin: 0 auto 10px;
        display: block;
        border-radius: 50%;
    }

    &__profile-title {
        font-size: 16px;
        font-weight: bold;
        text-align: center;
        margin: 10px 0 0;
        min-height: 1em;
    }
}
