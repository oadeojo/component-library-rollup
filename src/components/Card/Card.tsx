import classnames from 'classnames';
import React from 'react';

import styles from './styles.scss';

export interface Props {
    children?: any;
    headerImg?: string;
    title?: string;
}

const Card: React.StatelessComponent<Props> = (props) => {
    const { children, headerImg, title } = props,
        css = classnames(
            styles['card'],
            styles['card-container']
        );

    return (
        <div className={ css }>
            { (headerImg) && <img className={ styles['card__profile-img'] } src={ headerImg } /> }
            { (title) && <p className={ styles['card__profile-title'] }>{ title }</p> }
            { children }
        </div>
    );
};

Card.displayName = 'Card';

export default Card;
