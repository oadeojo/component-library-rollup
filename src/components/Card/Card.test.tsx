import { shallow } from 'enzyme';
import shallowToJson from 'enzyme-to-json';
import * as React from 'react';

import Card from './Card';

describe('Card', () => {
    it('should render with no props', () => {
        const output = shallow(
            <Card />
        );
        expect(shallowToJson(output)).toMatchSnapshot();
    });
    it('should render title', () => {
        const output = shallow(
            <Card title="My Card Title" />
        );
        expect(shallowToJson(output)).toMatchSnapshot();
    });
    it('should render a header image', () => {
        const output = shallow(
            <Card headerImg="plop.png" />
        );
        expect(shallowToJson(output)).toMatchSnapshot();
    });
});
