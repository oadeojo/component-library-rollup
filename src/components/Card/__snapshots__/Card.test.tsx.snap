// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render a header image 1`] = `
<div
  className="card card-container"
>
  <img
    className="card__profile-img"
    src="plop.png"
  />
</div>
`;

exports[`Card should render title 1`] = `
<div
  className="card card-container"
>
  <p
    className="card__profile-title"
  >
    My Card Title
  </p>
</div>
`;

exports[`Card should render with no props 1`] = `
<div
  className="card card-container"
/>
`;
